stable/pomerium/templates/proxy-deployment.yaml

```diff
@@ -53,41 +53,42 @@ spec:
         - name: COOKIE_SECRET
           valueFrom:
             secretKeyRef:
-              name: {{ template "pomerium.fullname" . }}
+              name: {{ if not .Values.config.existingSecret }}{{ template "pomerium.fullname" . }}{{ else }}{{ .Values.config.existingSecret }}{{ end }}
               key: cookie-secret
-        - name: SHARED_SECRET
+        - name: COOKIE_SECRET
           valueFrom:
             secretKeyRef:
-              name: {{ template "pomerium.fullname" . }}
-              key: shared-secret             
+              name: {{ if not .Values.config.existingSecret }}{{ template "pomerium.fullname" . }}{{ else }}{{ .Values.config.existingSecret }}{{ end }}
+              key: cookie-secret
         - name: AUTHENTICATE_SERVICE_URL
           value: {{ .Values.proxy.authenticateServiceUrl }}
         - name: AUTHORIZE_SERVICE_URL
           value: {{ .Values.proxy.authorizeServiceUrl }}
-{{- if .Values.config.policyFile }}
+{{- if .Values.proxy.authenticateUseInternalUrl }}
+        - name: AUTHENTICATE_INTERNAL_URL
+          value: {{ template "pomerium.fullname" . }}-{{ .Values.authenticate.name }}
+{{- end }}
+{{- if .Values.proxy.authorizeUseInternalUrl }}
+        - name: AUTHORIZE_INTERNAL_URL
+          value: {{ template "pomerium.fullname" . }}-{{ .Values.authorize.name }}
+{{- end }}
         - name: POLICY_FILE
           value: /etc/pomerium/policy.yaml
-{{- end }}
-{{- if .Values.config.policy}}
         - name: POLICY
           value: {{ .Values.config.policy }}
-{{- end }}
         - name: CERTIFICATE
           valueFrom:
             secretKeyRef:
-              name: {{ template "pomerium.fullname" . }}
-              key: proxy-cert
+              name: {{ if not .Values.config.existingSecret }}{{ template "pomerium.fullname" . }}{{ else }}{{ .Values.config.existingSecret }}{{ end }}
         - name: CERTIFICATE_KEY
           valueFrom:
             secretKeyRef:
-              name: {{ template "pomerium.fullname" . }}
-              key: proxy-key
```

Comment:
@desimone you mistakenly removed the key here

Reply:
Thank you!